Marlin/src/inc/SanityCheck.h

```diff
@@ -811,12 +811,10 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
 #if HAS_FILAMENT_SENSOR
   #if !PIN_EXISTS(FIL_RUNOUT)
     #error "FILAMENT_RUNOUT_SENSOR requires FIL_RUNOUT_PIN."
-  #elif NUM_RUNOUT_SENSORS > E_STEPPERS
-    #if HAS_PRUSA_MMU2
+  #elif HAS_PRUSA_MMU2 && NUM_RUNOUT_SENSORS != 1
       #error "NUM_RUNOUT_SENSORS must be 1 with MMU2 / MMU2S."
-    #else
-      #error "NUM_RUNOUT_SENSORS cannot exceed the number of E steppers."
-    #endif
+  #elif !(NUM_RUNOUT_SENSORS == 1 || NUM_RUNOUT_SENSORS == E_STEPPERS)
+    #error "NUM_RUNOUT_SENSORS must be either 1 or number of E steppers."
```

Comment:
Is this an existing limitation or a new one?

I assumed it was possible to have two extruders but only have a filament sensor on one of them.

Reply:
The logic has changed a bit, but the general limitation was already there before.
You can have one sensor on one extruder or sensors on all extruders (or multiple sensors wired parallel to one sensor input).
The change only effects you if you have 3 or more extruders. Here is the question how realistic is it to have sensors on only 2 out of 3 extruders vs. making a configuration mistake.

Reply:
Good question. Still, we allow for it because… maybe the third extruder is melting chocolate chips…?